### PR TITLE
refactor: Make ComponentSet.retrieve() Calls Consistently

### PR DESCRIPTION
**Pull request by klewis-sfdc**
### What does this PR do?
* Makes retrieve operations more consistent by 
** Setting the Api Version the same way
** Passing the connection to the Retrieve call the same way 

The two main Retrieve operations in the Extensions are when using a Retrieve command, and when loading the cache.
This PR makes the two retrieve calls more consistent, and is a useful item to have in place before exposing deploy and retrieve for scratch orgs (#4597).

### What issues does this PR fix or reference?
@W-12527686@

### Functionality Before
* API version was set in one instance, and was not set in the other instance
* Username was used in one instance, and the Connection was passed in the other instance
* The Retrieve Operation was passed back to the caller before being fully set up, resulting in difficult to debug stack traces

### Functionality After
* API version is set consistently
* Retrieve call uses the Connection in both places
* createRetrieveOperation is easier to debug with the included await

### CODE REVIEW

File: packages/salesforcedx-vscode-core/src/services/sdr/componentSetUtils.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { ConfigUtil } from '@salesforce/salesforcedx-utils-vscode/src';
```
Comment (gbockus-sf) [inline packages/salesforcedx-vscode-core/src/services/sdr/componentSetUtils.ts:1]: The above shouldn't include the src part of the path.  This is also why your int tests failed.

File: packages/salesforcedx-vscode-core/src/services/sdr/componentSetUtils.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  // If no user-configured Api Version is present, then get the version from the Org.
+  const orgApiVersion = await getOrgApiVersion();
+  components.apiVersion = orgApiVersion ?? components.apiVersion;
+}
+
+async function getOrgApiVersion(): Promise<string | undefined> {
+  const connection = await WorkspaceContext.getInstance().getConnection();
+  const apiVersion = connection.getApiVersion();
+  return apiVersion ? String(apiVersion) : undefined;
+}
```
Comment (gbockus-sf) [inline packages/salesforcedx-vscode-core/src/services/sdr/componentSetUtils.ts:26]: jest test incoming for these?

**Review by gbockus-sf — COMMENTED** [review]
Changes look good.  Probably gonna be ugly writing a jest test for baseDeployRetrieve & metadataCacheService, but the componentSetUtils test should be fairly straightforward 👍

File: packages/salesforcedx-vscode-core/src/context/workspaceContext.ts [hunk 3]
```diff
@@ -43,7 +43,7 @@ export class WorkspaceContext {
   }
 
   protected async handleCliConfigChange(orgInfo: OrgUserInfo) {
-    workspaceContextUtils.setupWorkspaceOrgType(orgInfo.username).catch(e =>
+    setupWorkspaceOrgType(orgInfo.username).catch(e =>
```
Comment (gbockus-sf) [inline packages/salesforcedx-vscode-core/src/context/workspaceContext.ts:46]: why'd you change this?

File: packages/salesforcedx-vscode-core/test/jest/services/sdr/componentSetUtils.test.ts [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    //   const configApiVersion = '45.0';
+    //   sb.stub(ConfigUtil, 'getUserConfiguredApiVersion').returns(
+    //     configApiVersion
+    //   );
+
+    //   await executor.run({ data: {}, type: 'CONTINUE' });
+    //   const components = executor.lifecycle.doOperationStub.firstCall.args[0];
+
+    //   expect(components.apiVersion).to.equal(getComponentsResult.apiVersion);
+    // });
```
Comment (gbockus-sf) [inline packages/salesforcedx-vscode-core/test/jest/services/sdr/componentSetUtils.test.ts:97]: Is this test invalid?

Comment (gbockus-sf) [reply packages/salesforcedx-vscode-core/test/jest/services/sdr/componentSetUtils.test.ts:97]: seems invalid and should be removed.

Comment (klewis-sfdc) [reply packages/salesforcedx-vscode-core/test/jest/services/sdr/componentSetUtils.test.ts]: Yep - removed.

File: packages/salesforcedx-vscode-core/src/context/workspaceContext.ts (hunk 3 above)
Comment (klewis-sfdc) [reply packages/salesforcedx-vscode-core/src/context/workspaceContext.ts:46]: Was probably a misguided dev attempt while trying to get the unit tests running.  I conflate this one with the workspaceContext instance issue sometimes I think.  I set it back.

File: packages/salesforcedx-vscode-core/test/jest/services/sdr/componentSetUtils.test.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    onDidCreate: jest.fn(),
+    onDidDelete: jest.fn()
+  };
+
+  beforeEach(() => {
+    workspaceContextGetConnectionMock = jest
+      .spyOn(WorkspaceContext.prototype, 'getConnection')
+      .mockResolvedValue(mockConnection);
+    createFileSystemWatcherMock = (vscode.workspace
+      .createFileSystemWatcher as any).mockReturnValue(mockWatcher);
```
Comment (gbockus-sf) [inline packages/salesforcedx-vscode-core/test/jest/services/sdr/componentSetUtils.test.ts:47]: interesting this mock was necessary

**Review by gbockus-sf — APPROVED** [review]
Code changes look good. 
Able to run a retrieve from the context menu ✅
Able to run a deploy with conflicts enabled ✅
Able to run a deploy and detect conflict ✅

File: packages/salesforcedx-vscode-core/test/jest/services/sdr/componentSetUtils.test.ts (hunk 5 above)
Comment (klewis-sfdc) [reply packages/salesforcedx-vscode-core/test/jest/services/sdr/componentSetUtils.test.ts:47]: @gbockus-sf it is not needed anymore now that this branch is parented by #4682.  I'm going to remove it before merging this FYI.

**Pull request merged**